src/utils/debouncer.js

```diff
@@ -0,0 +1,37 @@
+import Vue from 'vue';
+
+export default class DebounceCounter extends Vue {
+  constructor(delay = 200) {
+    super({
+      data() {
+        return { flag: false };
+      },
+    });
+    this.count = 0;
+    this.delay = delay;
+    this.timeout = null;
+  }
+
+  inc() {
+    this.count += 1;
+    if (this.count > 0) {
+      this._toggleFlag(true);
+    }
+  }
+
+  dec() {
+    this.count -= 1;
+    if (this.count === 0) {
```

Comment:
Should we clamp this to make sure it never goes below 0?

Reply:
Sure, done.